api/transport/request.go

```diff
@@ -60,3 +64,25 @@ type Request struct {
 
 // Encoding represents an encoding format for requests.
 type Encoding string
+
+// ValidateRequest validates the given request. An error is returned if the
+// request is invalid.
```

Comment:
Suggest adding something like "This func should be called from transports, including 3rd party ones".